udf-compiler/src/test/scala/com/nvidia/spark/OpcodeSuite.scala

```diff
@@ -45,19 +42,1834 @@ class OpcodeSuite extends FunSuite {
+    val dataset2 = List(2L).toDS()
+    val result2 = dataset2.withColumn("new", u('value))
+    val ref2 = dataset2.withColumn("new", lit(8L))
+    checkEquiv(result2, ref2)
+  }
+
+  test("conditional longs - AND(LTE,LT)") {
+    val myudf: Long => Long = { x =>
+      val t =
+        if (x <= 1L && x < 3L) {
+          2L
+        } else {
+          3L
+        }
+      t + 5L
+    }
+    val u = makeUdf(myudf)
+    val dataset = List(1L).toDS()
+    val result = dataset.withColumn("new", u('value))
+    val ref = dataset.withColumn("new", lit(7L))
+    checkEquiv(result, ref)
+    val dataset2 = List(2L).toDS()
+    val result2 = dataset2.withColumn("new", u('value))
+    val ref2 = dataset2.withColumn("new", lit(8L))
+    checkEquiv(result2, ref2)
+  }
+
+  test("conditional longs - AND(GT,GT)") {
+    val myudf: Long => Long = { x =>
+      val t =
+        if (x > 1L && x > 3L) {
+          2L
+        } else {
+          3L
+        }
+      t + 5L
+    }
+    val u = makeUdf(myudf)
+    val dataset = List(0L).toDS()
+    val result = dataset.withColumn("new", u('value))
+    val ref = dataset.withColumn("new", lit(8L))
+    checkEquiv(result, ref)
+    val dataset2 = List(4L).toDS()
+    val result2 = dataset2.withColumn("new", u('value))
+    val ref2 = dataset2.withColumn("new", lit(7L))
+    checkEquiv(result2, ref2)
+  }
+
+
+  test("conditional longs - AND(GT,GTE)") {
+    val myudf: Long => Long = { x =>
+      val t =
+        if (x > 1L && x >= 3L) {
+          2L
+        } else {
+          3L
+        }
+      t + 5L
+    }
+    val u = makeUdf(myudf)
+    val dataset = List(0L).toDS()
+    val result = dataset.withColumn("new", u('value))
+    val ref = dataset.withColumn("new", lit(8L))
+    checkEquiv(result, ref)
+    val dataset2 = List(3L).toDS()
+    val result2 = dataset2.withColumn("new", u('value))
+    val ref2 = dataset2.withColumn("new", lit(7L))
+    checkEquiv(result2, ref2)
+  }
+
+
+  test("conditional longs - OR(GT,GTE)") {
+    val myudf: Long => Long = { x =>
+      val t =
+        if (x < 1L && x <= 3L) {
+          2L
+        } else {
+          3L
+        }
+      t + 5L
+    }
+    val u = makeUdf(myudf)
+    val dataset = List(0L).toDS()
+    val result = dataset.withColumn("new", u('value))
+    val ref = dataset.withColumn("new", lit(7L))
+    checkEquiv(result, ref)
+    val dataset2 = List(4L).toDS()
+    val result2 = dataset2.withColumn("new", u('value))
+    val ref2 = dataset2.withColumn("new", lit(8L))
+    checkEquiv(result2, ref2)
+  }
+
+
+  test("loops - fallback test") {
+    val myudf: (Int, Int) => Int = (a,b) => {
+      var myVar : Int = 0
+      for (indexVar <- a to b){
+        myVar = myVar + 1
+      }
+      myVar
+    }
+    val u = makeUdf(myudf)
+    val dataset = List((1,5)).toDF("x","y")
+    val result = dataset.withColumn("new", u(col("x"),col("y")))
+    val ref = dataset.withColumn("new", lit(5))
+    checkEquiv(result, ref)
+  }
+
+  // new tests for string ops
```

Comment:
nit, could just say "string ops test" all tests are new to this repo :)

Reply:
yes, updated.